mam/api/api.h

```diff
@@ -294,21 +294,23 @@ retcode_t mam_api_deserialize(trit_t const *const buffer, size_t const buffer_si
  *
  * @param api - The API [in]
  * @param filename - The file name where to serialize the API into [in]
+ * @param encryption_key - The encryption key [in] (optional - can set to trits_null())
 
  *
  * @return return code
  */
-retcode_t mam_api_save(mam_api_t const *const api, char const *const filename);
+retcode_t mam_api_save(mam_api_t const *const api, char const *const filename, trits_t encryption_key);
```

Comment:
I think encryption/decryption should be done on serialize/deserialize instead. Save/load would be encrypted since they use it. But we must imagine other storage layer being used and the serialized trits should also be encrypted.